Lib/test/test__colorize.py

```diff
@@ -18,38 +17,93 @@ def tearDownModule():
+                with unittest.mock.patch("os.environ", {"NO_COLOR": "1"}):
+                    self.assertEqual(_colorize.can_colorize(), False)
+                with unittest.mock.patch("os.environ", {"NO_COLOR": "1", "PYTHON_COLORS": "1"}):
+                    self.assertEqual(_colorize.can_colorize(), True)
+                with unittest.mock.patch("os.environ", {"FORCE_COLOR": "1"}):
+                    self.assertEqual(_colorize.can_colorize(), True)
+                with unittest.mock.patch("os.environ", {"FORCE_COLOR": "1", "NO_COLOR": "1"}):
+                    self.assertEqual(_colorize.can_colorize(), False)
+                with unittest.mock.patch("os.environ", {"FORCE_COLOR": "1", "PYTHON_COLORS": "0"}):
+                    self.assertEqual(_colorize.can_colorize(), False)
+
+                isatty_mock.return_value = False
+                with unittest.mock.patch("os.environ", {}):
+                    self.assertEqual(_colorize.can_colorize(), False)
+
+            # If virtual terminal sequences are not supported
+            supports_vt_mock.return_value = False
             with (unittest.mock.patch("os.isatty") as isatty_mock,
-                  unittest.mock.patch("sys.flags", flags),
+                  unittest.mock.patch("sys.flags", unittest.mock.MagicMock(ignore_environment=False)),
                   unittest.mock.patch("_colorize.can_colorize", ORIGINAL_CAN_COLORIZE)):
                 isatty_mock.return_value = True
-                with unittest.mock.patch("os.environ", {'TERM': 'dumb'}):
+                with unittest.mock.patch("os.environ", {}):
+                    self.assertEqual(_colorize.can_colorize(), False)
+                with unittest.mock.patch("os.environ", {"TERM": "dumb"}):
                     self.assertEqual(_colorize.can_colorize(), False)
-                with unittest.mock.patch("os.environ", {'PYTHON_COLORS': '1'}):
+                with unittest.mock.patch("os.environ", {"PYTHON_COLORS": "1"}):
                     self.assertEqual(_colorize.can_colorize(), True)
-                with unittest.mock.patch("os.environ", {'PYTHON_COLORS': '0'}):
+                with unittest.mock.patch("os.environ", {"PYTHON_COLORS": "0"}):
                     self.assertEqual(_colorize.can_colorize(), False)
-                with unittest.mock.patch("os.environ", {'NO_COLOR': '1'}):
+                with unittest.mock.patch("os.environ", {"NO_COLOR": "1"}):
                     self.assertEqual(_colorize.can_colorize(), False)
-                with unittest.mock.patch("os.environ",
-                                         {'NO_COLOR': '1', "PYTHON_COLORS": '1'}):
+                with unittest.mock.patch("os.environ", {"NO_COLOR": "1", "PYTHON_COLORS": "1"}):
                     self.assertEqual(_colorize.can_colorize(), True)
-                with unittest.mock.patch("os.environ", {'FORCE_COLOR': '1'}):
+                with unittest.mock.patch("os.environ", {"FORCE_COLOR": "1"}):
```

Comment:
Please revert these formatting changes, it makes it harder to review.

Reply:
Done